crates/solver/src/settlement_submission/gelato.rs

```diff
@@ -1,2 +1,114 @@
-#[allow(dead_code)]
+//! Gelato settlement submission strategy.
+
 mod trampoline;
+
+use self::trampoline::Trampoline;
+use super::SubmissionError;
+use crate::settlement::Settlement;
+use anyhow::{anyhow, Context as _, Result};
+use contracts::GPv2Settlement;
+use ethcontract::{Account, H256};
+use shared::{
+    ethrpc::Web3,
+    gelato_api::{GelatoClient, TaskId, TaskState},
+};
+use std::time::Duration;
+use web3::types::TransactionReceipt;
+
+/// A Gelato submitter.
+pub struct GelatoSubmitter {
+    web3: Web3,
+    client: GelatoClient,
+    trampoline: Trampoline,
+    poll_interval: Duration,
+}
+
+impl GelatoSubmitter {
+    pub async fn new(web3: Web3, settlement: GPv2Settlement, client: GelatoClient) -> Result<Self> {
+        let trampoline = Trampoline::initialize(settlement).await?;
+        Ok(Self {
+            web3,
+            client,
+            trampoline,
+            poll_interval: Duration::from_secs(5),
```

Comment:
This might be too much if used on networks with lower block time? Might be useful to add it to configuration so we can tweak for each network individually if needed.

Reply:
Good point, added configuration option.